Conversions/ArbitraryBase.js

```diff
@@ -30,21 +46,11 @@ const convertArbitraryBase = (stringInBaseOne, baseOneCharacters, baseTwoCharact
   let stringInBaseTwo = ''
   const stringTwoBase = baseTwoCharacters.length
   while (value > 0) {
-    const remainder = value % stringTwoBase
-    stringInBaseTwo = baseTwoCharacters.charAt(remainder) + stringInBaseTwo
-    value /= stringTwoBase
+    const [divisionResult, remainder] = floorDiv(value, stringTwoBase)
+    stringInBaseTwo = baseTwoCharacters[remainder] + stringInBaseTwo
```

Comment:
Wait. Taking a closer look, this is still linear time (making the entire algo quadratic time in the number of digits), since it always prepends the character, which is a linear time operation since the string has to be copied. Ideally, the digits should be pushed to an array, which should then be `reverse`d and finally `join`ed.

Note: This is just a suggestion for further improvement, your PR is good to be merged as is.

Reply:
Fixed. I also added an alternate version that uses [BigInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt), which automatically uses floor division and will work for arbitrarily long input strings.